lib/parse.js

```diff
@@ -41,8 +42,29 @@ async function parse (path, $refs, options) {
 
     return parser.result;
   }
-  catch (e) {
-    return Promise.reject(e);
+  catch (ex) {
+    if (!("error" in ex)) {
+      throw ex;
+    }
+    else if (!options.failFast) {
+      $ref.pathType = ex.plugin.name;
+      file.data = null;
+      $ref.value = null;
+
+      const errors = $ref.errors || ($ref.errors = []);
+
+      if (ex.error instanceof StoplightParserError) {
```

Comment:
Suggest `ParserError` instead of `StoplightParserError`.

Reply:
We've got both `ParserError` and `StoplightParserError`